pephub/routers/api/v1/schemas.py

```diff
@@ -168,26 +170,45 @@ async def create_schema_for_namespace_by_file(
         )
 
 
-@schemas.get("/{namespace}/{schema}", response_model=SchemaGetResponse)
+@schemas.get("/{namespace}/{schema}", response_model=Union[SchemaGetResponse, dict])
 async def get_schema(
     namespace: str,
     schema: str,
     agent: PEPDatabaseAgent = Depends(get_db),
+    return_type: Optional[Literal["yaml", "json"]] = "json",
 ):
     try:
         schema_dict = agent.schema.get(namespace=namespace, name=schema)
+    except SchemaDoesNotExistError:
+        raise HTTPException(
+            status_code=404, detail=f"Schema {schema}/{namespace} not found."
+        )
+    if return_type == "yaml":
+
         info = agent.schema.info(namespace=namespace, name=schema)
         return SchemaGetResponse(
             schema=yaml.dump(schema_dict),
             description=info.description,
             last_update_date=info.last_update_date,
             submission_date=info.submission_date,
         )
+    else:
+        return schema_dict
+
```

Comment:
Can we check if they sent a valid return_type?

```python
if return_type not in VALID_RETURN_TYPES:
    raise HTTPException("Not a valid return type")
```

Reply:
I feel like they always be correct, as I am converting dict to str and vice versa. But we can add it if needed

Reply:
Hmm true. Only concern is it could be confusing to the user if I say `return_type=rubber-duck` and the it returns... Does `rubber-duck` do anything? But you are correct its already there to catch things, so fine leaving it 👍🏼